exercises/Anthony_Lim/ex02_oo_basics/src/shape_sorter.cpp

```diff
@@ -0,0 +1,94 @@
+//This is the introduction exercise 2. The work has been completed by Anthony Lim 
+/*
+------------------------------------------------------------------------------
+Include statements:
+
+ 	internal headers - same project
+ 	internal headers - different project
+ 	3rd part headers
+	system
+-------------------------------------------------------------------------------
+*/
+
+#include "shape_sorter.h" 
+#include<string> 
+
+/*
+Aurthor: Anthony Lim
+data: 10/01/2017
+*/
+
+/*
+This body file is for the shape_sorter class 
+*/
+void Shape_Sorter::swap(Shape* &shape1, Shape* &shape2)
+{
+	Shape* tmp1 = shape1;
+	Shape* tmp2 = shape2;
+	shape1 = tmp2;
+	shape2 = tmp1;
+}
+	
+void Shape_Sorter::get_Match_Shape(std::string this_shape)
+{
+	for(int j =0;j<m_shapes.size();j++)
```

Comment:
Consider using a range-based for-loop. See [here](http://en.cppreference.com/w/cpp/language/range-for)